### DISCUSSION TITLE
".do4" files not showing in "Race Results" screen

### DISCUSSION THREAD

**Discussion by kylemchristian02** [JohnStrunk/wahoo-results#575] [Q&A] [answered]
We use a OneDrive on our timing computer to write the files to so that the scoring computer(meet manager) can pull the times down and score them. I first tried to point to the OneDrive and get the .do4 files from there but they never populated in the Race Results window. Since that wasn't working, I decided to pull a copy down to the local machine and try it. Still no race results files populated in the Race Results window.

**Comment by JohnStrunk** [comment]
That's a strange one... Here's some troubleshooting ideas:

- Sorry if this one is too basic... gotta start w/ the easy stuff--- Make sure you've set the correct directory to look for the race results. (In the "Directories tab", use the "Browse" button on the right side to set the directory.
- My next thought is that one or more of the do4 files might be corrupted. I have a couple samples in [testdata](https://github.com/JohnStrunk/wahoo-results/tree/master/testdata) you could try. Create an empty directory, put one of them into that new directory, and see if it is recognized.

I don't know whether using OneDrive (or remote file systems in general) will work, but local directories certainly should. Let me know what you find out.

**Comment by kylemchristian02** [comment, marked as answer]
So it actually ended up being the do4 file I was using. When I created the .do4 files, I didn't have event numbers in the CTS Dolphin SW. Therefore, when the .do4 files were written, there were no event numbers written. Through testing, it seems like you have a check in the parser that looks for event number and heat number and even race number and if either of those three are blank in the .do4 file, it will not import it into the system.

**Reply by kylemchristian02** [reply]
After testing it on my local desktop, I also tested it and it worked correctly on OneDrive so it isn't a file system issue.

**Reply by JohnStrunk** [reply]
Glad you got it figured out. In the past, when I didn't have events loaded in Dolphin, I thought it just marked it as event 1, heat 1 in the file.

Could you send me (or attach here) an example file or two so I can update the parser?

**Reply by kylemchristian02** [reply]
[005-000-001A-0006.do4.txt](https://github.com/user-attachments/files/15571327/005-000-001A-0006.do4.txt)

**Reply by JohnStrunk** [reply]
Thank you!
